src/Data/Characters/Beidou/index.tsx

```diff
@@ -1,185 +1,271 @@
+import { cond, trans } from '../../SheetUtil'
+import CharacterSheet, { conditionalHeader, ICharacterSheet, normalSrc, talentTemplate } from '../CharacterSheet'
+import { customDmgNode, dataObjForCharacterSheet, dmgNode, shieldNode, shieldNodeTalent } from '../dataUtil'
+import { banner, burst, c1, c2, c3, c4, c5, c6, card, passive1, passive2, passive3, skill, thumb, thumbSide } from './assets'
+import data_gen_src from './data_gen.json'
+import skillParam_gen from './skillParam_gen.json'
+
+const key: CharacterKey = "Beidou"
+const elementKey: ElementKey = "electro"
+const [tr, trm] = trans("char", key)
+const data_gen = data_gen_src as CharacterData
+
+let a = 0, s = 0, b = 0
+const datamine = {
+  normal: {
+    hitArr: [
+      skillParam_gen.auto[a++],
+      skillParam_gen.auto[a++],
+      skillParam_gen.auto[a++],
+      skillParam_gen.auto[a++],
+      skillParam_gen.auto[a++],
+    ]
+  },
+  charged: {
+    spinningDmg: skillParam_gen.auto[a++],
+    finalDmg: skillParam_gen.auto[a++],
+    stamina: skillParam_gen.auto[a++][0],
+    duration: skillParam_gen.auto[a++][0],
+  },
+  plunging: {
+    dmg: skillParam_gen.auto[a++],
+    low: skillParam_gen.auto[a++],
+    high: skillParam_gen.auto[a++],
+  },
+  skill: {
+    shieldHp_: skillParam_gen.skill[s++],
+    shieldFlat: skillParam_gen.skill[s++],
+    dmgBase: skillParam_gen.skill[s++],
+    onHitDmgBonus: skillParam_gen.skill[s++], //DMG bonus on hit taken
+    cd: skillParam_gen.skill[s++][0],
+  },
+  burst: {
+    burstDmg: skillParam_gen.burst[b++],
+    lightningDmg: skillParam_gen.burst[b++],
+    damageReduction: skillParam_gen.burst[b++],
+    duration: skillParam_gen.burst[b++][0],
+    cd: skillParam_gen.burst[b++][0],
+    energyCost: skillParam_gen.burst[b++][0],
+  },
+  //pasive 1: 2, //additional targets for lightning arc
+  ascension4: {
+    normalDmg_: skillParam_gen.passive2[0][0], //Same value for all 3
+    chargeDmg_: skillParam_gen.passive2[0][0],
+    attackSpeed: skillParam_gen.passive2[0][0],
+  },
+  constellation1: {
+    shieldHp_: skillParam_gen.constellation1[0],
+  },
+  constellation4: {
+    skillDmg: skillParam_gen.constellation4[0],
+    duration: skillParam_gen.constellation4[1],
+  },
+  constellation6: {
+    electroResShred_: -1 * skillParam_gen.constellation6[0],
+  },
+} as const
+
+//Toggable stuff: 
+// A4: Unleashing <b>Tidecaller</b> with its maximum DMG Bonus
+// C6: During the duration of <b>Stormbreaker</b>
+
+const [condC6Path, condC6] = cond(key, "Constellation6")
+const [condA4Path, condA4] = cond(key, "Ascension4")
+
+const nodeC3 = greaterEq(input.constellation, 3, 3)
+const nodeC5 = greaterEq(input.constellation, 5, 3)
+
+const skillDmgOneHit = datamine.skill.dmgBase.map((dmg,i) => dmg + datamine.skill.onHitDmgBonus[i])
+const skillDmgTwoHits = datamine.skill.dmgBase.map((dmg,i) => dmg + 2 * datamine.skill.onHitDmgBonus[i])
+
+const electroShieldStr_ = { "customBonus": { "shield_": percent(1.5) } }
```

Comment:
this would make it additive instead of multiplicative?

Reply:
It's adding the multipliers that then get multiplied in the damage formula

`skillDmgOneHit ` ends up being `[121.6 + 160 , 130.72 + 172 , 139.84 + 184]` etc
`skillDmgOneHit ` ends up being `[[121.6 + 2*160 , 130.72 + 2*172 , 139.84 + 2*184]` etc

The numbers also match current GO

Current:
![image](https://user-images.githubusercontent.com/10319468/158905880-b3183ee7-0cf8-4613-9d89-691e3ec98204.png)

Waverider:
![image](https://user-images.githubusercontent.com/10319468/158905987-5b2e4f24-6e51-41a3-9af5-332023688c39.png)

750.7 == 206+2*272

Also just realized that Waverider is rounding the cooldown from 7.5 to 8

Reply:
My comment is towards the `electroShieldStr_ `, looks like someone already added elemental shields, so we can fix it in post.

there is a `fixed` field in the `field` that can fix rounding